rp2040-hal/src/pwm/mod.rs

```diff
@@ -0,0 +1,655 @@
+//! Pulse Width Modulation (PWM)
+//!
+//! To access the PWM pins you must call the 'split' method on the PWM. This will return a
+//! `_____` struct with access to each PWM pin:
+//!
+//! ```no_run
+//! use rp2040_hal::{prelude::*, pwm::Slices};
+//!
+//!
+//! let mut pac = rp2040_pac::Peripherals::take().unwrap();
+//!
+//! // Init PWMs
+//! let pwm_slices = Slices::new(pac.PWM, &mut pac.RESETS);
+//!
+//! // Configure PWM4
+//! let mut pwm = pwm_slices.pwm4;
+//! pwm.set_ph_correct();
+//! pwm.enable();
+//! ```
+//!
+//! Once you have the PWM slice struct, you can add individual pins:
+//!
+//! ```no_run
+//! # use rp2040_hal::{prelude::*, gpio::Pins, sio::Sio, pwm::Slices};
+//! # let mut pac = rp2040_pac::Peripherals::take().unwrap();
+//! # let pwm_slices = Slices::new(pac.PWM, &mut pac.RESETS);
+//! # let mut pwm = pwm_slices.pwm4;
+//! # let mut pac = rp2040_pac::Peripherals::take().unwrap();
+//! #
+//! # let sio = Sio::new(pac.SIO);
+//! # let pins = Pins::new(
+//! #     pac.IO_BANK0,
+//! #     pac.PADS_BANK0,
+//! #     sio.gpio_bank0,
+//! #     &mut pac.RESETS,
+//! # );
+//! #
+//! use embedded_hal::PwmPin;
+//!
+//! // Use B channel (which outputs to GPIO 25)
+//! let mut channel = pwm.channel_b;
+//! let channel_pin = channel.output_to(pins.gpio25);
+//!
+//! // Set duty cycle
+//! channel.set_duty(0x00ff);
+//! channel.get_duty();
+//! channel.set_inverted(); // Invert the output
+//! channel.clr_inverted(); // Don't invert the output
+//! ```
+//!
+//! The following configuration options are also available:
+//!
+//! ```no_run
+//! # use rp2040_hal::{prelude::*, pwm::Slices};
+//! # let mut pac = rp2040_pac::Peripherals::take().unwrap();
+//! # let pwm_slices = Slices::new(pac.PWM, &mut pac.RESETS);
+//! # let mut pwm = pwm_slices.pwm4;
+//! pwm.set_ph_correct(); // Run in phase correct mode
```

Comment:
I think it would be good to show here how to bring the pwm slice into one of the other modes as well, since it's not obvious.
```rust
/// Set this channel 
let pwm: &mut hal::pwm::Slice<hal::pwm::Pwm4, hal::pwm::InputHighRunning> =
        &mut pwm_slices.pwm4.into_mode();
```